activerecord/lib/active_record/encryption/scheme.rb

```diff
@@ -87,8 +84,15 @@ def validate_credential(key, error_message = "is not configured")
           end
         end
 
+        def key_provider_from_config
```

Comment:
I suggest renaming to `default_key_provider`. `...from_config`  feels a bit ambiguous to me. This would also be consistent with private method `Encrytor#default_key_provider`.

Also, the implementation can be simplified to:

```ruby
 ActiveRecord::Encryption.key_provider
```

This is because, by convenience, there are top level getters for all the context properties. See `ActiveRecord::Encryption::Configurable`.

By the way, this is a good change. It's good that `Scheme` defaults to the `default_key_provider` if nothing else is provided. Without this change, it was working as expected because `Encryptor#decrypt` and `Encryptor#encrypt` provide a default value for this param, but that was kind of brittle 👍.